core/src/main/scala/io/aiven/guardian/kafka/Utils.scala

```diff
@@ -1,10 +1,26 @@
 package io.aiven.guardian.kafka
 
+import scala.annotation.tailrec
+
 import java.time.OffsetDateTime
+import java.time.format.DateTimeParseException
 
 object Utils {
+
+  private def parseToOffsetDateTime(string: String): Option[OffsetDateTime] =
+    try
+      Some(OffsetDateTime.parse(string))
+    catch {
+      case _: DateTimeParseException =>
+        None
+    }
+
+  @tailrec
   def keyToOffsetDateTime(key: String): OffsetDateTime = {
```

Comment:
This had to be changed to deal with the fact that we know have extensions `.json.gz` where as before we only had `.json`. The previous implementation would only drop the ending `.gz` from the key which means you still had `<ISO_DATE_TIME>.json`.

The nicest way I could find to solve this issue is to just recursively drop ending extensions between the period (`.`) until you happen to successfully parse the key as a date.